e2e/terraform/nomad-acls.tf

```diff
@@ -12,24 +12,20 @@ locals {
   nomad_env = "NOMAD_ADDR=https://${aws_instance.server.0.public_ip}:4646 NOMAD_CACERT=keys/tls_ca.crt NOMAD_CLIENT_CERT=keys/tls_api_client.crt NOMAD_CLIENT_KEY=keys/tls_api_client.key"
 }
 
+# write the bootstrap token to the keys/ directory (where the ssh key is) so
+# that we can read it into the data.local_file later. If not set, ensure that
+# it's empty.
 resource "null_resource" "bootstrap_nomad_acls" {
   depends_on = [module.nomad_server]
   triggers = {
-    script = data.template_file.bootstrap_nomad_script.rendered
+    script = templatefile("${local.nomad_env} ./scripts/bootstrap-nomad.sh", {})
   }
 
   provisioner "local-exec" {
-    command = data.template_file.bootstrap_nomad_script.rendered
+    command = templatefile("${local.nomad_env} ./scripts/bootstrap-nomad.sh", {})
```

Comment:
The `templatefile` function takes a path, whereas this is a rendered string that was originally the rendered _output_ of a `template_file` block. I _think_ we can probably do without the `templatefile` command here entirely if we verify that the `local.nomad_env` is properly interpolated.

(Plus the `triggers` block above becomes useless once we're no longer responding to that render event.)

Reply:
I made a correction in [d467da7](https://github.com/hashicorp/nomad/pull/19313/commits/d467da720c830ad7fd8c6a307b48699c007c4fe9) which I think makes sense.